src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -71,19 +74,31 @@ public CommandManagerWindow(DriverCommandManager commandManager) {
 		content.add(bntGiveCommand, c);
 
 		JButton btnClearObservers = new JButton("Delete observers");
-		btnClearObservers.addActionListener((ActionEvent e) -> this.deleteObservers());
+		btnClearObservers.addActionListener((ActionEvent e) -> this.deleteObservers(btnClearObservers));
 		c.fill = GridBagConstraints.BOTH;
 		c.weightx = 1;
 		c.gridx = 0;
 		c.weighty = 1;
 		content.add(btnClearObservers, c);
+
+		JButton btnRunCommand = new JButton("Run command");
+		btnRunCommand.addActionListener((ActionEvent e) -> this.runCommand());
+		c.fill = GridBagConstraints.BOTH;
+		c.weightx = 1;
+		c.gridx = 0;
+		c.weighty = 1;
+		content.add(btnRunCommand, c);
 	}
 
 	private void clearCommand() {
 		commandManager.clearCurrentCommand();
 		updateCurrentCommandField();
 	}
 
+	private void runCommand() {
+		commandManager.runCurrentCommand();
```

Comment:
As a result, of refactor, now CommandManagerWindow is indirectly dependent on DriverManager. Try to fix it. Suggestion: try creating interface(s).